gnome-initial-setup/pages/branding-welcome/gis-branding-welcome-page.c

```diff
@@ -0,0 +1,289 @@
+
+struct _GisBrandingWelcomePagePrivate {
+  gchar *title;
+  gchar *description;
+  gchar *logo_path;
+  gchar *skip_pages;
+};
+typedef struct _GisBrandingWelcomePagePrivate GisBrandingWelcomePagePrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE (GisBrandingWelcomePage, gis_branding_welcome_page, GIS_TYPE_PAGE);
+
+#define OBJ(type,name) ((type)gtk_builder_get_object(GIS_PAGE(page)->builder,(name)))
+#define WID(name) OBJ(GtkWidget*,name)
+
+/*
+ * CONFIG_FILEPATH points to /etc/eos-branding/gnome-initial-setup/pages.conf, which is a
+ * file containing groups of key-value pairs under a group called "Welcome", defining the
+ * usual parameters needed to render the "branding-welcome" page.
+ *
+ * Among those parameters, the only required one is the "title", meaning that the page will
+ * be forgiving about the rest of them not being there, rendering the title and any other
+ * optional field that it's there, if present.
+ *
+ * Fields:
+ *   - title (required): string containing a short string to show as title.
+ *   - description (optional): string containing long text, likely to be wrapped.
+ *   - logo (optional): absolute path to the file with a logo for the brand.
+ *   - skip (optional): comma-separated list of PAGE IDs the branded g-i-s should skip.
+ *
+ * For example, this is how this file would look on a branded image defining a title,
+ * a description, a logo and skipping the "language" and "keyboard" pages.
+ *
+ *   [Welcome]
+ *   title=A title to be shown at the top
+ *   description=A long description that will be shown at the bottom of this page, right below the branded logo (if any), explaining what the branded edition is about.
+ *   logo=/path/to/the/image/with/the/logo.png
+ *   skip=language,keyboard
+ */
+#define CONFIG_FILEPATH CONFIGDIR "/eos-branding/gnome-initial-setup/pages.conf"
+
+static void
+load_css_overrides (GisBrandingWelcomePage *page)
+{
+  g_autoptr(GtkCssProvider) provider;
+  g_autoptr(GFile) file;
+  g_autoptr(GError) error = NULL;
+
+  provider = gtk_css_provider_new ();
+  file = g_file_new_for_uri ("resource:///org/gnome/initial-setup/branding-welcome-page.css");
+  gtk_css_provider_load_from_file (provider, file, &error);
+
+  if (error != NULL)
+    {
+      g_warning ("Unable to load CSS overrides for the branding-welcome page: %s",
+                 error->message);
+      return;
+    }
+
+  gtk_style_context_add_provider_for_screen (gtk_widget_get_screen (GTK_WIDGET (page)),
+                                             GTK_STYLE_PROVIDER (provider),
+                                             GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
+}
+
+static gboolean
+read_config_file (GisBrandingWelcomePage *page)
+{
+  GisBrandingWelcomePagePrivate *priv = NULL;
+  g_autoptr(GKeyFile) keyfile = NULL;
+  g_autoptr(GError) error = NULL;
+  g_auto(GStrv) groups = NULL;
+  gboolean ret = FALSE;
+  gsize n_groups;
+  int i;
+
+  if (!g_file_test (CONFIG_FILEPATH, G_FILE_TEST_EXISTS))
+    goto out;
```

Comment:
There's no need to check for this  separately -- `g_key_file_load_from_file()` will fail when it does not exist. You can check the error if you don't want to warn in that case.